pulseapi/tests.py

```diff
@@ -44,11 +48,40 @@ def setup_groups():
     staff.save()
 
 
-def setup_entries(test):
-    test.entries = [EntryFactory() for i in range(2)]
-    for entry in test.entries:
+def setup_entries(test, creator_users):
+    test.entries = []
+    test.creators = []
+
+    for i in range(2):
+        entry = EntryFactory()
         entry.save()
 
+        # Create a simple creator that has no profile
+        creators = [CreatorFactory()]
+        if creator_users and len(creator_users) > i:
+            # If we were passed in users, create a creator attached to a user
+            creators.append(CreatorFactory(user=creator_users[i]))
+        for creator in creators:
+            creator.save()
+            # Connect the creator with the entry
+            OrderedCreatorRecord(entry=entry, creator=creator).save()
+
+        test.creators.extend(creators)
+        test.entries.append(entry)
+
+
+def setup_users_with_profiles(test):
+    users = [EmailUserFactory() for i in range(3)]
+    for user in users:
+        user.save()
+        profile = UserProfileFactory(
```

Comment:
hm, this is an interesting problem. Right now we automatically build a user profile when a never-before-seen user logs in through gauth, do we want to add to that so that profiles are also build when a new user is built outside of social authentication?

Reply:
probably not needed.